### PR TITLE
Refactoring FormFields and ValidateForm to be separate components

### PR DESCRIPTION
**Pull request by B1s9l**
# Pull Requests Checklist

Assignee:

- \[x\] code commented?
- \[x\] deactivated things not needed? (e.g. other features which only were prepared)
- \[x\] alerts that this is todo?
- \[x\] test run localhost?
- \[x\] tested manually all features and edge cases?
- \[x\] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?

Reviewer:

- \[ \] correct branch?
- \[ \] are there any code smells?
- \[ \] did pipeline pass?
- \[ \] is the code well documented?
- \[ \]tested manually all features and edge cases?

### CODE REVIEW

File: src/components/popups/TeamSetting.tsx [hunk 1]
```diff
@@ -147,11 +87,13 @@ const TeamSettings = ({ isOpen, onClose, onEdit, setIsLeave }) => {
         },
       });
       onEdit();
+      setIsLoading(false);
       DeactivateEditMode();
     } catch (error) {
       setError("Something went wrong. Please try again");
       console.error("Error while updating Team", handleError(error));
     }
+    setIsLoading(false);
```
Comment (5v3nn) [inline src/components/popups/TeamSetting.tsx:96]: `setError` false?

File: src/components/popups/TeamSetting.tsx [hunk 2]
```diff
@@ -206,6 +148,13 @@ const TeamSettings = ({ isOpen, onClose, onEdit, setIsLeave }) => {
     }
   };
 
+  const getAllErrorMessages = () => {
+    const fieldErrors = Object.values(errors).filter((error) => error);
+    if (generalError) fieldErrors.push(generalError);
+
+    return fieldErrors;
+  };
+
```
Comment (5v3nn) [inline src/components/popups/TeamSetting.tsx:157]: seems like a general function that can be used in other places as well. maybe appropriate to add to `FormField` component?

Comment (B1s9l) [reply src/components/popups/TeamSetting.tsx]: Done

File: src/components/popups/TeamSetting.tsx (hunk 1 above)
Comment (B1s9l) [reply src/components/popups/TeamSetting.tsx:97]: Need further explanation what is asked for?

File: src/components/utilities/ErrorMessages.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+const ErrorMessages = ({ errors, generalError }) => {
+  const getErrorMessages = () => {
+    const fieldErrors = Object.values(errors).filter((error) => error);
+    if (generalError) {
+      fieldErrors.push(generalError);
+    }
+    return fieldErrors;
+  };
+
```
Comment (Monti934) [inline src/components/utilities/ErrorMessages.tsx:12]: please add newline before return statement

**Review by Monti934 — CHANGES_REQUESTED** [review]

File: src/components/views/Registration.tsx [hunk 4]
```diff
@@ -155,12 +106,8 @@ const Registration = () => {
...
                 </Button>
               </div>
+              <ErrorMessages errors={errors} generalError={generalError} />
             </div>
-            {getAllErrorMessages().map((error, index) => (
-              <div key={index} className="error-message">
-                {error}
-              </div>
-            ))}
             <label className="register message">
```
Comment (Monti934) [inline src/components/views/Registration.tsx:111]: please put the error message outside of the container like Login.tsx and add a margin for consistent styling

File: src/styles/popups/InspectTask.scss [hunk 5]
```diff
@@ -90,8 +63,3 @@
     margin-left: auto;
   }
 }
-
-.bts {
-  font-size: medium;
-  width: 15%;
-}
```
Comment (Monti934) [inline src/styles/popups/InspectTask.scss:97]: when you remove my styling, please add new styling. Now several buttons look off, like "Close" on create Team or "Submit" for comments in InspectTask

**Review by Monti934 — CHANGES_REQUESTED** [review]
